docs/adr/adr_database-history.md

```diff
@@ -0,0 +1,150 @@
+
+*Advantages*
+
+- change date and user are recorded
+- simple, especially querying (select by highest value of `effective` date column)
+- separation of data and history
+- can be combined with database triggers to record any change on DB level (no change in application code necessary then)
+- history tables should be insert-only; they cannot be modified then
+
+*Drawbacks*
+
+- extensibility costs: every change in the original table requires an update in the corresponding history table
+- the state of foreign key fields (i.e. tables that a history table depends on) has to be recorded as well to still be correct later
+
+### Insert-only tables
+
+Whenever a row would change, it is marked as inactive and copied into a new, active row.
+
+*Advantages*
+
+- no additional tables needed
+
+*Drawbacks*
+
+- no separation of responsibilities (relevant data vs. history)
+- it's not possible to refer to such table entries using foreign keys (no referential integrity)
+
+### Snapshots
+
+Make copies of the database state at repeated times and archive them.
+
+*Advantages*
+
+- copying is simple
+- no extension of business logic required
+
+*Drawbacks*
+
+- analysing is complicated: DB schema might have changed over time
+- storage needs to be defined: in DB or as separate file
+- precise changes are not recorded
+
+### Event sourcing
+
+Every change to the state of an application is captured in an event object. These event objects are themselves stored in the sequence they were applied for the same lifetime as the application state itself. The event log can be used to reconstruct past states.
+
+*Advantages*
+
+- powerful, scalable concept
+- helpful for auditing and debugging (replaying events in a test environment)
+
+*Drawbacks*
+
+- complex extension of codebase required (twice for dropapp and v2 back-end)
+- new developers might need to familiarize themselves with the concept first
+
+## Decision
+
+There are several popular audit logging techniques, but none of them serve every purpose. The most effective ones are often expensive, resource intensive, or performance degrading. Others are cheaper in terms of resources but are either incomplete, cumbersome to maintain, or require a sacrifice in design quality.
+
+I propose to use history tables. It meets the requirements for simplicity and for knowing historical state. Since we don't have a lot of tables to be shadowed (stock, products, locations, people, transactions), the maintenance cost is low.
+
+Quoting the first reference below,
+
+> If you only want to store logs for a few tables, shadow tables may be the most convenient option.
+
+## Consequences
+
+See advantages/drawbacks above.
+
+We have to think about
+
+- whether we still want `created_at`, `modified_at`, etc. fields in tables that are shadowed
```

Comment:
If we remove them, we might have to change code in the php app. Therefore, I would keep them and remove them once we are rid of the php app :).